ctwrap/modules/adiabatic_flame.py

```diff
@@ -94,27 +88,33 @@ def run(name,
     if loglevel > 0:
         f.show_solution()
     msg = '    {0:s}: mixture-averaged flamespeed = {1:7f} m/s'
-    print(msg.format(name, f.u[0]))
+    print(msg.format(name, f.velocity[0]))
 
-    out[name + '<mix>'] = pd.DataFrame(
-        np.vstack([f.grid, f.u, f.V, f.T, f.density, f.X]).T, columns=keys)
+    if name is not 'defaults':
+        group = name + "<mix>"
+    else:
+        group = "mix"
+    out[group] = f
 
     # Solve with multi-component transport properties
     f.transport_model = 'Multi'
     f.solve(loglevel)  # don't use 'auto' on subsequent solves
     if loglevel > 0:
         f.show_solution()
     msg = '    {0:s}: multi-component flamespeed  = {1:7f} m/s'
-    print(msg.format(name, f.u[0]))
+    print(msg.format(name, f.velocity[0]))
 
-    out[name + '<multi>'] = pd.DataFrame(
-        np.vstack([f.grid, f.u, f.V, f.T, f.density, f.X]).T, columns=keys)
+    if name is not 'defaults':
+        group = name + "<multi>"
+    else:
+        group = "multi"
+    out[group] = f
 
     return out
```

Comment:
Per prior discussion: shouldn't `write_hdf` be called from within this function?

Reply:
Actually, this needs a separate `save` method to properly execute in parallel.